File: python-exercicios/lista02/OperaMatriz.py
#*******************************************************************************)
#*** Autor: Aron Daniel Lopes												****)
#*** Data: 06/05/2011														****)
#*** Este programa preenche uma matrix 5x5 com valores e realiza diversas  ****)
#*** operacos de soma e exibicao de linhas, colunas, diagonais e a matriz ****)
#*** inteira.																****)
#******************************************************************************)


#&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&*)
#&&&& Trecho de funcoes e procedimentos do programa &&&&*)
#&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&*)


# Esta funcao retorna a soma de determinada linha *)
def RetornaSomaLinha(linha):
	soma = 0
	for coluna in range(5):
		soma =soma + matriz_[linha-1][coluna]
	return soma



# Esta funcao retorna a soma de determinada coluna *)
def RetornaSomaColuna(coluna):
	soma=0
	for linha in range(5):
		soma=soma + matriz_[linha][coluna-1]
	return soma



# Esta funcao retorna a soma da diagonal principal *)
def RetornaSomaDiagonalP():
	soma = 0
	for linha in range(5):
		coluna=linha
		soma=soma + matriz_[linha][coluna]
	return soma



# Esta funcao retorna a soma da diagonal secundaria *)
def RetornaSomaDiagonalS():
	soma = 0
	coluna = 4
	for linha in range(5):
		soma = soma + matriz_[linha][coluna]
		coluna=coluna-1
	return soma

# Esta funcao retorna a soma de uma matriz*)
def RetornaSomaMatriz():
	soma = 0
	for linha in range(5):
		for coluna in range(5):
			soma=soma + matriz_[linha][coluna]
	return soma

#========================================================================*)
# =======================  Escopo principal =============================*)
#========================================================================*)


print "Este programa realiza diversas operacoes com uma matriz. "
	
# O for abaixo le os elementos e os insere na matriz. *)
matriz_ = []
for linha in range(5):
	matriz_.append([0]*5)
	for coluna in range(5):
		print ""
		matriz_[linha][coluna]=input("Digite o valor que ira compor o elemento da linha " + str(linha) + " e coluna " + str(coluna) + " :")

# For abaixo imprime a matriz digitada. *)
print ""
print "A matriz digitada:"
for linha in range(5):
	for coluna in range(5):
		print str(matriz_[linha][coluna]) + "     ",
	print ""

print "A soma da quarta linha: " + str(RetornaSomaLinha(4))
print "A soma da segunda coluna: " + str(RetornaSomaColuna(2))
print "A soma da diagonal principal: " + str(RetornaSomaDiagonalP())
print "A soma da diagonal secundaria: " + str(RetornaSomaDiagonalS())
print "A soma de todos os elementos da matriz: " + str(RetornaSomaMatriz())


#========================================================================*)

File: fortran-exericios/lista02/OperaMatriz.f
(*******************************************************************************)
(**** Autor: Aron Daniel Lopes												****)
(**** Data: 06/05/2011														****)
(**** Este programa preenche uma matrix 5x5 com valores e realiza diversas  ****)
(**** operações de soma e exibição de linhas, colunas, diagonais e a matriz ****)
(**** inteira.																****)
(*******************************************************************************)

Program OperaMatriz;

(*--------------------------------------------*)

type
	matriz = Array[1..5] of Array[1..5] of Integer;
	
(*--------------------------------------------*)

(*%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*)

var
	matriz_: Matriz;
	linha, coluna: Integer;

(*%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*)

(*&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&*)
(*&&&& Trecho de funções e procedimentos do programa &&&&*)
(*&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&&*)


(* Esta função retorna a soma de determinada linha *)
Function RetornaSomaLinha(linha: Integer): Integer;

var
	soma: Integer;
	
begin
	soma:=0;
	for coluna:=1 to 5 do
	begin
		soma:=soma + matriz_[linha][coluna];
	end;
	RetornaSomaLinha:=soma;
end;



(* Esta função retorna a soma de determinada coluna *)
Function RetornaSomaColuna(coluna: Integer): Integer;

var
	soma: Integer;
	
begin
	soma:=0;
	for linha:=1 to 5 do
	begin
		soma:=soma + matriz_[linha][coluna];
	end;
	RetornaSomaColuna:=soma;
end;



(* Esta função retorna a soma da diagonal principal *)
Function RetornaSomaDiagonalP(): Integer;

var
	soma: Integer;
	
begin
	soma:=0;
	for linha:=1 to 5 do
	begin
		coluna:=linha;
		soma:=soma + matriz_[linha][coluna];
	end;
	RetornaSomaDiagonalP:=soma;
end;



(* Esta função retorna a soma da diagonal secundária *)
Function RetornaSomaDiagonalS(): Integer;

var
	soma: Integer;
	
begin
	soma:=0;
	coluna:=5;
	for linha:=1 to 5 do
	begin
		soma:=soma + matriz_[linha][coluna];
		coluna:=coluna-1;
	end;
	RetornaSomaDiagonalS:=soma;
end;

(* Esta função retorna a soma de uma matriz*)
Function RetornaSomaMatriz(): Integer;

var
	soma: Integer;
	
begin
	soma:=0;
	for linha:=1 to 5 do
	begin
		for coluna:=1 to 5 do
		begin
			soma:=soma + matriz_[linha][coluna];
		end;
	end;
	RetornaSomaMatriz:=soma;
end;

(*========================================================================*)
(* =======================  Escopo principal =============================*)
(*========================================================================*)

begin

	WriteLn('Este programa realiza diversas operações com uma matriz.');
	
	(* O for abaixo le os elementos e os insere na matriz. *)
	for linha:=1 to 5 do
	begin
		for coluna:=1 to 5 do
		begin
			WriteLn('Digite o valor que irá compor o elemento da linha ', linha ,' e coluna ', coluna, ' :');
			ReadLn(matriz_[linha][coluna]);
		end;
	end;
	
	(* For abaixo imprime a matriz digitada. *)
	WriteLn('A matriz digitada:');
	for linha:=1 to 5 do
	begin
		for coluna:=1 to 5 do
		begin
			Write(matriz_[linha][coluna], '     ');
		end;
		WriteLn();
	end;
	
	WriteLn('A soma da quarta linha: ', RetornaSomaLinha(4)); 
	WriteLn('A soma da segunda coluna: ', RetornaSomaColuna(2)); 
	WriteLn('A soma da diagonal principal: ', RetornaSomaDiagonalP()); 
	WriteLn('A soma da diagonal secundária: ', RetornaSomaDiagonalS()); 
	WriteLn('A soma de todos os elementos da matriz: ', RetornaSomaMatriz()); 
	
end.

(*========================================================================*)